### PR TITLE
Upgrade Maven resolver related dependencies used by bootstrap

### LINKED ISSUE #3539: QuarkusTestExtension does not support -ntp in Maven 3.6.1
**Describe the bug**

If I run Maven commands using `-ntp`, an option added in Maven 3.6.1, I get an error during Quarkus tests.

**Expected behavior**

Just runs.

**Actual behavior**

The test crashes with

```
org.junit.jupiter.api.extension.TestInstantiationException: TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class […]
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestInstanceFactory(ClassBasedTestDescriptor.java:298)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateTestClass(ClassBasedTestDescriptor.java:273)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:70)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateAndPostProcessTestInstance(ClassBasedTestDescriptor.java:256)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$2(ClassBasedTestDescriptor.java:250)
	at java.util.Optional.orElseGet(Optional.java:267)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$3(ClassBasedTestDescriptor.java:249)
	at org.junit.jupiter.engine.execution.TestInstancesProvider.getTestInstances(TestInstancesProvider.java:29)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$prepare$0(TestMethodTestDescriptor.java:106)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:105)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$1(NodeTestTask.java:107)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:107)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:75)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:150)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.ExceptionInInitializerError
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:118)
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:56)
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver$Builder.build(MavenArtifactResolver.java:101)
	at io.quarkus.bootstrap.BootstrapClassLoaderFactory.newDeploymentClassLoader(BootstrapClassLoaderFactory.java:314)
	at io.quarkus.test.junit.QuarkusTestExtension.createQuarkusBuildClassLoader(QuarkusTestExtension.java:291)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:78)
	at io.quarkus.test.junit.QuarkusTestExtension.createTestInstance(QuarkusTestExtension.java:343)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestInstanceFactory(ClassBasedTestDescriptor.java:283)
	... 49 more
Caused by: java.lang.IllegalStateException: Failed to parse Maven command line arguments
	at io.quarkus.bootstrap.resolver.maven.MavenRepoInitializer.<clinit>(MavenRepoInitializer.java:101)
	... 57 more
Caused by: org.apache.commons.cli.UnrecognizedOptionException: Unrecognized option: -ntp
	at org.apache.commons.cli.Parser.processOption(Parser.java:383)
	at org.apache.commons.cli.Parser.parse(Parser.java:210)
	at org.apache.commons.cli.Parser.parse(Parser.java:88)
	at org.apache.maven.cli.CLIManager.parse(CLIManager.java:159)
	at io.quarkus.bootstrap.resolver.maven.MavenRepoInitializer.<clinit>(MavenRepoInitializer.java:99)
	... 57 more
```

**To Reproduce**

Given a sample app,

```bash
DOCKER_BUILDKIT=1 docker build --progress plain -f src/main/docker/Dockerfile.test -t test .
```

```Dockerfile
# syntax=docker/dockerfile:1.1.1-experimental
FROM quay.io/quarkus/centos-quarkus-maven:19.1.1
COPY src /project/src
COPY pom.xml /project
USER root
RUN cp -rv ~quarkus/.m2 ~root/.m2
RUN --mount=type=cache,target=/root/.m2/repository,sharing=private mvn -f /project -ntp package
```

crashes. If `-ntp` is removed, the build works.

**Environment**

Ubuntu, Docker 19.03.1

**Additional context**

I suspect https://github.com/quarkusio/quarkus/blob/08180a306b88e010ac54c44dddbd01d8ad7569ac/independent-projects/bootstrap/pom.xml#L29 is to blame here.

### LINKED ISSUE #7458: Unable to build gradle project in Java 11 + smallrye amqp extension
**Describe the bug**
When starting a build task with gradle in Java 11, an error occured :

Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.smallrye.reactivemessaging.deployment.SmallRyeReactiveMessagingProcessor#build threw an exception: java.lang.NullPointerException
        at org.apache.commons.lang3.SystemUtils.isJavaVersionAtLeast(SystemUtils.java:1626)
        at org.apache.commons.lang3.ClassUtils.isAssignable(ClassUtils.java:690)
        at io.smallrye.reactive.messaging.MediatorConfigurationSupport.validateSubscriber(MediatorConfigurationSupport.java:100)
        at io.smallrye.reactive.messaging.MediatorConfigurationSupport.validate(MediatorConfigurationSupport.java:77)
        at io.quarkus.smallrye.reactivemessaging.deployment.QuarkusMediatorConfigurationUtil.create(QuarkusMediatorConfigurationUtil.java:76)
        at io.quarkus.smallrye.reactivemessaging.deployment.SmallRyeReactiveMessagingProcessor.build(SmallRyeReactiveMessagingProcessor.java:226)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

**To Reproduce**
1. Create a quarkus project
2. Add smallrye amqp extension in the build.gradle
3. Create a classe with @Incoming annotation
4. Launch build or quarkusBuild task

This error is due to the Gradle plugin which included an old lib org.apache.commons:commons-lang3:3.5 rather than a 3.9 :
classpath
\--- io.quarkus:io.quarkus.gradle.plugin:1.3.0.Alpha2
     \--- gradle.plugin.io.quarkus:quarkus-gradle-plugin:1.3.0.Alpha2
          +--- io.quarkus:quarkus-bootstrap-core:1.3.0.Alpha2
          |    +--- org.ow2.asm:asm:7.3.1
          |    +--- org.apache.maven:maven-embedder:3.5.4
          |    |    +--- org.apache.maven:maven-settings:3.5.4
          |    |    |    \--- org.codehaus.plexus:plexus-utils:3.1.0
          |    |    +--- org.apache.maven:maven-settings-builder:3.5.4
          |    |    |    +--- org.apache.maven:maven-builder-support:3.5.4
          |    |    |    |    \--- org.apache.commons:commons-lang3:3.5

**Configuration**
```build.gradle
implementation "io.quarkus:quarkus-smallrye-reactive-messaging-amqp"

```

**Environment (please complete the following information):**
 - Output of `java -version`: 11
 - GraalVM version (if different from Java): 11
 - Quarkus version : 1.2.1.Final or 1.3.0.Alpha2 

**Additional context**
Gradle Plugin version : 1.2.1.Final or 1.3.0.Alpha2

### PR DESCRIPTION
**Pull request by aloubyansky**
to those used by/compatible with Maven 3.6.3 release

Fixes #3539
Fixes #7458

### CODE REVIEW

File: extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java [hunk 1]
```diff
@@ -265,7 +265,10 @@ private void visitAnnotation(final DotName clazz, final IndexView index, final M
                 visitAnnotation(annotation, index, deps, visited, ret);
             }
         }
-        ret.add(index.getClassByName(clazz));
+        final ClassInfo classInfo = index.getClassByName(clazz);
+        if (classInfo != null) {
+            ret.add(classInfo);
+        }
```
Comment (gsmet) [inline extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java]: @geoand can you check that change?

**Review by gsmet — APPROVED** [review]
Looks good to me. Asked @geoand to check the Spring change.

File: extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java (hunk 1 above)
Comment (aloubyansky) [reply extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java]: I saw an NPE there.

Comment (aloubyansky) [reply extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java]: I saw an NPE there.

Comment (aloubyansky) [reply extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java]: Well, not at this line exactly.

**Review by geoand — APPROVED** [review]
Weird NPE, have never seen it to there. Check can't hurt of course. I'll keep an eye out for any issues that might be related to Spring DI.

**Comment by aloubyansky** [comment]
If you want to reproduce it @geoand, take this commit an undo that particular change.

**Comment by geoand** [comment]
I'll do that now

**Comment by geoand** [comment]
@aloubyansky which test failed?  I can't seem to reproduce

**Comment by aloubyansky** [comment]
I think it was in spring-di deployment.

**Comment by aloubyansky** [comment]
or in the IT. Sorry, don't remember for sure. Do you want me to reproduce it for you? I even remember logging the annotation. I think it was javax.annotation.Nonnull.

**Comment by geoand** [comment]
I tried the deployment module and all Spring ITs and nothing tripped...

**Comment by aloubyansky** [comment]
Ok, I'll try again.

**Comment by geoand** [comment]
Oh, nevermind, I was able to reproduce it now

**Comment by geoand** [comment]
OK, fix is good and should have been there in the first place :).

Thanks @aloubyansky

**Pull request merged**